libs/providers/flagd/src/lib/service/common/grpc-util.ts

```diff
@@ -0,0 +1,16 @@
+import { ClientReadableStream } from '@grpc/grpc-js';
+
+export const closeStreamIfDefined = (stream: ClientReadableStream<unknown> | undefined) => {
+  /**
+   * cancel() is necessary to prevent calls from hanging the process, so we need to we need to remove all the
+   * handlers, and add a no-op for 'error' so we can cancel without bubbling up an exception
+   */
+  if (stream) {
+    stream.removeAllListeners();
+    stream.on('error', () => {
+      // swallow errors after closed
+    });
+    stream.cancel();
+    stream.destroy();
```

Comment:
This is the actual fix, which I've extracted to a util and made use of in both grpc connections.